firebase-firestore/src/main/java/com/google/firebase/firestore/local/SQLiteRemoteDocumentCache.java

```diff
@@ -66,6 +69,41 @@ public MaybeDocument get(DocumentKey documentKey) {
         .firstValue(row -> decodeMaybeDocument(row.getBlob(0)));
   }
 
+  @Override
+  public Map<DocumentKey, MaybeDocument> getAll(Iterable<DocumentKey> documentKeys) {
+    List<Object> args = new ArrayList<>();
+    for (DocumentKey key : documentKeys) {
+      args.add(EncodedPath.encode(key.getPath()));
+    }
+
+    Map<DocumentKey, MaybeDocument> results = new HashMap<>();
+    Iterator<DocumentKey> keyIter = documentKeys.iterator();
+    while (keyIter.hasNext()) {
+      // Make sure each key has a corresponding entry, which is null in case the document is not
```

Comment:
Perhaps there's a better way to achieve this?

Reply:
Not sure.  Part of me would rather we do the db lookups first and then only insert nulls for the missing items. It feels like it matches the intention better, but it's probably a little extra code, and I don't know which way is actually more efficient.